openedx/core/lib/edx_api_utils.py

```diff
@@ -36,30 +38,37 @@ def get_edx_api_data(api_config, user, resource, querystring=None, cache_key=Non
         return no_data
 
     if cache_key:
+        cache_key = '{}.{}'.format(cache_key, resource_id) if resource_id else cache_key
+
         cached = cache.get(cache_key)
-        if cached is not None:
+        if cached:
             return cached
 
     try:
         jwt = get_id_token(user, api_config.OAUTH2_CLIENT_NAME)
         api = EdxRestApiClient(api_config.internal_api_url, jwt=jwt)
-    except Exception:  # pylint: disable=broad-except
+    except:  # pylint: disable=bare-except
         log.exception('Failed to initialize the %s API client.', api_config.API_NAME)
         return no_data
 
     try:
         querystring = {} if not querystring else querystring
-        response = getattr(api, resource).get(**querystring)
-        results = response.get('results', no_data)
-        page = 1
-        next_page = response.get('next', None)
-        while next_page:
-            page += 1
-            querystring['page'] = page
-            response = getattr(api, resource).get(**querystring)
-            results += response.get('results', no_data)
-            next_page = response.get('next', None)
-    except Exception:  # pylint: disable=broad-except
+        response = getattr(api, resource)(resource_id).get(**querystring)
+
+        if resource_id:
+            results = response
+        else:
+            results = response.get('results', no_data)
```

Comment:
#nitpick, The logic in the `else` statement looks like can be re-used and can be easier to read if it's in a function. Can you please wrap it in a private function?